ls1intum/Artemis
# [Bugfix] Do not update score if the test weight is zero

### Checklist

### Motivation and Context


Fixes #3414. An internal error happens when the student's score is updated with a test weight of 0. This can happen if you set the test weight of visible tests to 0 and set an invisible test to > 0.

### Description

Add a check to the test weight and skip the student score update if it is 0.

### Steps for Testing


1. Log in to Artemis
2. Navigate to Course Administration
3. Navigate to a programming exercise and click on Grading
4. Set the test weight to 0 for all of the tests
5. For the last test, set the visibility to "Never" and set the test weight to something greater than 0
6. Click Trigger All
7. Wait a minute and check that no error has been thrown

### Test Coverage
